app/integration/runtime/src/main/java/io/syndesis/integration/runtime/handlers/DataMapperStepHandler.java

```diff
@@ -74,17 +73,15 @@ public boolean canHandle(Step step) {
      * @param dataSources
      * @return
      */
-    private Optional<ProcessorDefinition<?>> handleJsonTypeSourceProcessor(ProcessorDefinition<?> route, List<Map<String, Object>> dataSources) {
+    private void addJsonTypeSourceProcessor(ProcessorDefinition<?> route, List<Map<String, Object>> dataSources) {
         List<String> jsonTypeSourceIds = dataSources.stream()
                                                     .filter(s -> ATLASMAP_JSON_DATA_SOURCE.equals(s.get("jsonType")) && "SOURCE".equals(s.get("dataSourceType")))
                                                     .map(s -> Optional.ofNullable(s.get("id")).map(Object::toString).orElse(""))
```

Comment:
wondering if this could be a filter, or an empty id has a meaning ?

Reply:
good point! added the filter